quantum/via.c

```diff
@@ -195,24 +189,96 @@ bool process_record_via(uint16_t keycode, keyrecord_t *record) {
+// If a keyboard is using the default Core modules, it does not need to be overridden,
+// the VIA keyboard definition will have matching channel/IDs.
+//
+// If a keyboard has some extra custom values, then via_custom_value_command_kb() can be
+// overridden to handle the extra custom values, leaving via_custom_value_command() to
+// handle the custom values for Core modules.
+//
+// If a keyboard has custom values and code that are overlapping with Core modules,
+// then via_custom_value_command() can be overridden and call the same functions
+// as the default implementation, or do whatever else is required.
+//
 // DO NOT call raw_hid_send() in the override function.
-__attribute__((weak)) void raw_hid_receive_kb(uint8_t *data, uint8_t length) {
+//
+
+// This is the default handler for "extra" custom values, i.e. keyboard-specific custom values
+// that are not handled by via_custom_value_command().
+__attribute__((weak)) void via_custom_value_command_kb(uint8_t *data, uint8_t length) {
+    // data = [ command_id, channel_id, value_id, value_data ]
     uint8_t *command_id = &(data[0]);
-    *command_id         = id_unhandled;
+    // Return the unhandled state
+    *command_id = id_unhandled;
 }
 
-// VIA handles received HID messages first, and will route to
-// raw_hid_receive_kb() for command IDs that are not handled here.
-// This gives the keyboard code level the ability to handle the command
-// specifically.
+// This is the default handler for custom value commands.
+// It routes commands with channel IDs to command handlers as such:
+//
+//      id_qmk_backlight_channel    ->  via_qmk_backlight_command()
+//      id_qmk_rgblight_channel     ->  via_qmk_rgblight_command()
+//      id_qmk_rgb_matrix_channel   ->  via_qmk_rgb_matrix_command()
+//      id_qmk_audio_channel        ->  via_qmk_audio_command()
 //
-// raw_hid_send() is called at the end, with the same buffer, which was
-// possibly modified with returned values.
+__attribute__((weak)) void via_custom_value_command(uint8_t *data, uint8_t length) {
+    // data = [ command_id, channel_id, value_id, value_data ]
+    uint8_t *channel_id = &(data[1]);
+
+#if defined(BACKLIGHT_ENABLE)
+    if (*channel_id == id_qmk_backlight_channel) {
+        via_qmk_backlight_command(data, length);
+        return;
+    }
+#endif // BACKLIGHT_ENABLE
+
+#if defined(RGBLIGHT_ENABLE)
+    if (*channel_id == id_qmk_rgblight_channel) {
+        via_qmk_rgblight_command(data, length);
+        return;
+    }
+#endif // RGBLIGHT_ENABLE
+
+#if defined(RGB_MATRIX_ENABLE)
+    if (*channel_id == id_qmk_rgb_matrix_channel) {
+        via_qmk_rgb_matrix_command(data, length);
+        return;
+    }
+#endif // RGBLIGHT_ENABL
+
+#if defined(AUDIO_ENABLE)
+    if (*channel_id == id_qmk_audio_channel) {
+        via_qmk_audio_command(data, length);
+        return;
+    }
+#endif // AUDIO_ENABLE
+
+    *channel_id = *channel_id; // force use of variable
```

Comment:
Guessing this was here when each of the above conditionals were disabled.
```suggestion
    (void)channel_id;
```